src/network/hub.rs

```diff
@@ -777,6 +779,15 @@ impl HubConn {
 
         Ok(())
     }
+
+    fn send_free_joints(&self) -> Result<()> {
+        let obj_joints = storage::read_free_joints().expect("send free joint failed");
```

Comment:
change to `let joints = storage::read_free_joints().context("send free joint failed")?;`